### ISSUE TITLE
Automerge doesn't work if the CI first fails but fixed in another update

### ISSUE THREAD

**Issue by Skaronator** [renovateapp/renovate#1381]
I've setup Renovate to automerge directly to the master if the CI build doesn't fail. Now it tried to prepare a update for `amazon-cognito-identity-js` but the CI build fail so it create a PR for it. Now `amazon-cognito-identity-js` push another update which fixes the builds again (_yeah_) but instead of automerge to the master it just update the PR.

Here what happend:
1. Current master branch
`"amazon-cognito-identity-js": "1.29.0"`

2. Broken Update -> Renovate create PR for:
`"amazon-cognito-identity-js": "1.30.0",` <- This is the broken version

3. AWS published a new (fixed) version:
`"amazon-cognito-identity-js": "1.31.0"`
 CI is now successfully but renovate update the PR instead of automerge it.

![image](https://user-images.githubusercontent.com/2886913/34862639-c2dee162-f76b-11e7-8817-d77ca2e8686d.png)

**Comment by rarkins** [comment]
Do you mean you have configured it for `branch-push`? If so then what you're seeing is deliberate behaviour, although I am open to changing it.

- If `branch-push` automerging is enabled, but the branch fails tests then we raise a PR (which I think is the right thing to do)
- If a PR exists - must have been because of a test failure previously - then we skip any automerging by branch and instead update the PR, leave it to a human

In your preferred approach (I think) you would want it still do the automerge the branch once tests pass again, e.g. a third step above and instead of leaving it to a human it would force push the update to `master`, delete the branch, and in doing so the PR would get automatically closed. Is that right?

**Comment by Skaronator** [comment]
Yeah it should merge the PR once the tests pass or since I use `branch-push` it should close the PR and directly push to master. IMO, what do you think?

Here is the whole config:
```
 "extends": [
    "config:base",
    ":automergeMinor",
    ":automergeBranchPush"
  ]
```

**Comment by rarkins** [comment]
I think if we do a check that there's still only one commit (i.e. make sure no human has been editing the branch) then it might be OK. If a human needed to manually fix an upgrade I think it would be confusing and perhaps risky to automerge it the second it passes tests - we should never automerge human-edited branches.

**Comment by rarkins** [comment]
Here is the current logic: https://github.com/renovateapp/renovate/blob/dba574950b09ced45ae67d87f2d70e245b3b0abc/lib/workers/branch/automerge.js#L10-L13

**Comment by Skaronator** [comment]
Yeah maybe change it to something like this: (pseudo code)
```JS
if (existingPr) {
  if (branchStatus === 'success')
  {
    const countCommits = await ....;
    if (countCommits === 1)
    {
      //Close the PR!
      await platform.mergeBranch(config.branchName, config.automergeType);
    }
    return 'automerge aborted - Human commits detected';
  }
  return 'automerge aborted - PR exists';
}
```

**Comment by rarkins** [comment]
In summary: if branch-x automerge and the branch is green and only one commit, then automerge regardless of whether a PR exists or not.

**Comment by Skaronator** [comment]
Yup
